Automattic/pocket-casts-ios
# Resume downloads on next launch after force quit

Fixes #472 

This PR adds a check for episodes in the `downloading` status that aren't actually downloading, and re-adds them to the queue. This is performed when other "resume download" logic is run in `checkForPendingAndAutoDownloads`

## To test

1. pull branch and build
2. start downloading an episode
3. once the download has started, force quit the app
4. reopen the app and browse to the episode again (or check the Profile -> Downloads section)
5. The episode should have auto resumed download
6. Ensure download only on wifi is enabled ( Profile -> Gear -> auto download -> Only on WiFi )
7. Force `isConnectedToWifi()` to return `false` and rebuild (or if there is some simulator setting I couldn't find that simulates a cell connection, do that)
8. Repeat the test plan, when you get to step 5 the episode SHOULD NOT resume download and you should see a wifi icon in the episode cell
9. Remove the forced cell connection behaviour and rebuild
10. When the app opens, return to the Profile -> Downloads list.
11. The episode should resume downloading.

NOTE: currently, the download resumes from 0% since there doesn't appear to be any download progress state saved. Also, it might not be desirable to resume from the existing file's position since this download could resume several days later which might cause issues with playback due to auto ad insertion.

NOTE 2: I tested in an iOS 14.1 simulator prior to applying this branch, and downloads didn't resume there either, so it doesn't appear that they ever resumed if no code changes have been made around this recently... or at least, its not the result of a fundamental change to iOS.

## Checklist
